AK/Vector.h

```diff
@@ -389,43 +425,73 @@ class Vector {
     }
 
     template<class... Args>
-    void empend(Args&&... args)
+    [[nodiscard]] bool try_empend(Args&&... args)
     {
-        grow_capacity(m_size + 1);
+        if (!try_grow_capacity(m_size + 1))
+            return false;
         new (slot(m_size)) T { forward<Args>(args)... };
         ++m_size;
+        return true;
     }
 
-    ALWAYS_INLINE void append(T&& value)
+    template<class... Args>
+    void empend(Args&&... args)
+    {
+        auto did_allocate = try_empend(forward<Args>(args)...);
+        VERIFY(did_allocate);
+    }
+
+    [[nodiscard]] ALWAYS_INLINE bool try_append(T&& value)
     {
-        grow_capacity(size() + 1);
+        if (!try_grow_capacity(size() + 1))
+            return false;
         new (slot(m_size)) T(move(value));
         ++m_size;
+        return true;
+    }
+
+    ALWAYS_INLINE void append(T&& value)
+    {
+        auto did_allocate = try_append(move(value));
+        VERIFY(did_allocate);
+    }
+
+    [[nodiscard]] ALWAYS_INLINE bool try_append(const T& value)
+    {
+        return try_append(T(value));
     }
 
     ALWAYS_INLINE void append(const T& value)
     {
         append(T(value));
     }
```

Comment:
For consistency, I think this could be changed to :
```
        auto did_allocate = try_append(value);
        VERIFY(did_allocate);
```

Reply:
Fixed